WeakAuras/WeakAuras.lua

```diff
@@ -3905,6 +3906,11 @@ function WeakAuras.PreAdd(data)
     WeakAuras.validate(data, default)
   end
 
+  local regionValidate = data.regionType and WeakAuras.regionTypes[data.regionType] and WeakAuras.regionTypes[data.regionType].validate
+  if regionValidate then
+    regionValidate(data)
```

Comment:
I'm surprised you don't do it only once with WeakAuras.Modernize, what is the reason for doing it here?